mods/stairs/init.lua

```diff
@@ -256,355 +373,425 @@ if replace then
 end
 
 
--- Stair/slab registration function.
--- Nodes will be called stairs:{stair,slab}_<subname>
+-- Stair/slab/stairi/stairo registration function.
+-- Nodes will be called stairs:{stair,slab,stairi,stairo}_<subname>
 
-function stairs.register_stair_and_slab(subname, recipeitem,
-		groups, images, desc_stair, desc_slab, sounds)
+function stairs.register_stair_and_slab_and_stairi_and_stairo(subname, recipeitem,
+		groups, images, desc_stair, desc_slab, desc_stairi, desc_stairo, sounds)
 	stairs.register_stair(subname, recipeitem, groups, images, desc_stair, sounds)
 	stairs.register_slab(subname, recipeitem, groups, images, desc_slab, sounds)
+    stairs.register_stairi(subname, recipeitem, groups, images, desc_stairi, sounds)
```

Comment:
I went through all of the code, and went tab-crazy.
hopefully all is properly indented now. I will resubmit in the morning (East Coast US time)
the code has changed quite a bit.